File: 백준/Gold/15980. 명상 방해꾼/명상 방해꾼.java
import java.io.*;
import java.util.StringTokenizer;
import java.util.ArrayList;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int[][] board = new int[N+1][M];
        int[] sum = new int[M]; // 각 초에서 스님 방해받은 정도

        for (int i = 1; i <= N; i++) {
            st = new StringTokenizer(br.readLine());
            char dir = st.nextToken().charAt(0);
            String arr = st.nextToken();
            for (int j = 0; j < M; j++) {
                int value = 0;
                if (arr.charAt(j) == '1') {
                    if (dir == 'L') {
                        value = -1;
                    } else if (dir == 'R') {
                        value = 1;
                    }
                    board[i][j] = value;
                    sum[j] += value;
                }
            }
        }

        int answerSound = 5000000;
        int answerIdx = 0;
        for (int i = 1; i <= N; i++) {
            int total = 0;
            int maxTotal = 0;

            for (int j = 0; j < M; j++) {
                total += sum[j] - board[i][j];
                maxTotal = Math.max(maxTotal, Math.abs(total));
            }

            if (answerSound > maxTotal) {
                answerSound = maxTotal;
                answerIdx = i;
            }
        }
        System.out.println(answerIdx);
        System.out.println(answerSound);

    }
}

File: 백준/Gold/15980. 명상 방해꾼/명상 방해꾼.py
N, M = map(int, input().split())
board = [[0] * M for _ in range(N+1)]
sum = [0] * (M)
for i in range(1, N+1):
    d, arr = input().split()
    for j in range(M):
        if arr[j] == '1':
            if d == 'L':
                value = -1
            else:
                value = 1

            sum[j] += value
            board[i][j] = value

answerTotal = 5000000
answerIdx = 0
for i in range(1, N+1):
    total = 0
    maxTotal = 0
    for j in range(M):
        total += sum[j] - board[i][j]
        maxTotal = max(maxTotal, abs(total))

    if answerTotal > maxTotal:
        answerTotal = maxTotal
        answerIdx = i

print(answerIdx)
print(answerTotal)